app/src/backend/environmentVariables/utils.ts

```diff
@@ -0,0 +1,18 @@
+import lodash from "lodash";
+import { EnvironmentVariableValue } from "./types";
+import Logger from "lib/logger";
+
+export const renderTemplate = (template: string, variables: Record<string, EnvironmentVariableValue>) => {
+  lodash.templateSettings.interpolate = /{{([\s\S]+?)}}/g;
+
+  const parsedVariable = Object.entries(variables).reduce((acc, [key, value]) => {
+    acc[key] = value.localValue ?? value.syncValue;
+    return acc;
+  }, {} as Record<string, string | number | boolean>);
+  try {
+    return lodash.template(template)(parsedVariable);
+  } catch (e) {
+    Logger.error("Error while rendering template");
+    return template;
+  }
+};
```

Comment:
We should use backbone here too as it is more extensible and we're already using it in mock server

https://github.com/requestly/requestly-mock-server/blob/1831d97d65eb335548090f0a0faa5b28936f6f90/src/core/utils/templating/index.ts#L8-L15